prow/cmd/phaino/main.go

```diff
@@ -112,6 +125,12 @@ func readPJs(ctx context.Context, jobs []string) (<-chan prowapi.ProwJob, <-chan
 			return
 		}
 		for _, j := range jobs {
+			select {
```

Comment:
This select statement needs to be combined with `ch <- *pj` (line 147) as one of its cases so that this goroutine is aborted when the context is cancelled and the goroutine is waiting to on `ch <- *pj` (the common case).